### PR TITLE
Fix #646

### LINKED ISSUE #646: Units per factory left is not working correctly
[Bug Video](https://youtu.be/4wK7iul9oWQ)
_Sorry for the bad quality... I'm not a professional video-recorder ^^_
If there are no units left to deploy, the limit seems to be back at the maximum. It shouldn't allow the user to increase the number and get an error afterwards

### PR DESCRIPTION
**Pull request by RoiEXLab**
This fixes #646

### CODE REVIEW

File: src/main/java/games/strategy/triplea/ui/PlacePanel.java [hunk 1]
```diff
@@ -88,9 +88,7 @@ public void territorySelected(final Territory territory, final MouseDetails e) {
       final UnitChooser chooser = new UnitChooser(units, Collections.emptyMap(), getData(),
           false, getMap().getUIContext());
       final String messageText = "Place units in " + territory.getName();
-      if (maxUnits[0] > 0) {
-        chooser.setMaxAndShowMaxButton(maxUnits[0]);
-      }
+      chooser.setMaxAndShowMaxButton(maxUnits[0]);
```
Comment (ron-murhammer) [inline src/main/java/games/strategy/triplea/ui/PlacePanel.java]: Not really following why this change is beneficial. Is there a reason you would want to set a max of 0 or less and show the max unit button?

Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/ui/PlacePanel.java]: If you watch the video I linked on #646, you will see, that if this method is not executed when maxunits = 0 if we open the frame, we can choose unlimited units just to see an error dialog in our face because that's an invalid thing to do...

Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/ui/PlacePanel.java]: To be fair, a ≥ 0 would probably also work...

Comment (ron-murhammer) [inline src/main/java/games/strategy/triplea/ui/PlacePanel.java]: Yeah I think >= 0 would be better here as maxUnits can be -1 if its unlimited and we probably don't want to show the max button in that case.

Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/ui/PlacePanel.java]: Since I'm currently in bed, and this is a very minor change, feel free to edit this snippet in the github online editor.

**Comment by simon33-2** [comment]
Shouldn't the dialog just not show if the max is zero?

**Comment by ron-murhammer** [comment]
Closing this for #1549
